src/main/java/com/spotify/reaper/ReaperApplication.java

```diff
@@ -62,25 +64,42 @@
 public class ReaperApplication extends Application<ReaperApplicationConfiguration> {
 
   static final Logger LOG = LoggerFactory.getLogger(ReaperApplication.class);
-
+  public final static UUID reaperInstanceId = UUID.randomUUID();
+  private static String reaperInstanceAddress;
```

Comment:
this should be a `static final` constant.

and assigned to in a static constructor, not the instance constructors.